homeassistant/components/light/decora.py

```diff
@@ -52,15 +53,21 @@ def __init__(self, device):
         """Initialize the light."""
         # pylint: disable=import-error
         import decora
-
         self._name = device['name']
         self._address = device['address']
         self._key = device["key"]
         self._switch = decora.decora(self._address, self._key)
-        self._switch.connect()
+        initial = time.time()
+        while True:
+            if time.time() - initial >= 10:
+                return None
+            try:
+                self._switch.connect()
+                break
+            except decora.decoraException:
+                pass
```

Comment:
I'm sorry for that hints. This change request is only for move this loop logic into `platform_setup` instead have a endless loop on `init` function.